From 8910ab1caa680ddef8b3fd7ca585e7a9f3c4c169 Mon Sep 17 00:00:00 2001
From: Luciano Santos <luc14n0@opensuse.org>
Date: Fri, 4 Aug 2023 22:21:20 -0300
Subject: [PATCH] build: Make installation of tests optional

Default to installing them, while letting people/distributors decide
what they want.
---
 meson_options.txt |  4 ++++
 src/meson.build   | 35 +++++++++++++++++++----------------
 2 files changed, 23 insertions(+), 16 deletions(-)

diff --git a/meson_options.txt b/meson_options.txt
index 82b6af0..a0683f9 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -40,3 +40,7 @@ option('polkit',
        type: 'feature',
        value: 'auto',
        description: 'PolKit support in daemon')
+option('installed_tests',
+       type : 'boolean',
+       value : 'true',
+       description : 'Install integration tests')
diff --git a/src/meson.build b/src/meson.build
index e4ed974..7aa1c66 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -146,20 +146,23 @@ if os_backend == 'linux' and gobject_introspection.found()
             )
     endforeach
 
-    install_data( [
-        'linux/integration-test.py',
-        'linux/output_checker.py',
-      ],
-      install_dir: get_option('prefix') / get_option('libexecdir') / 'upower'
-    )
-    install_subdir('linux/tests/',
-      install_dir: get_option('prefix') / get_option('libexecdir') / 'upower'
-    )
-
-    configure_file(
-      input: 'upower-integration.test.in',
-      output: 'upower-integration.test',
-      install_dir: get_option('datadir') / 'installed-tests' / 'upower',
-      configuration: cdata
-    )
+    installed_tests = get_option('installed_tests')
+    if installed_tests
+        install_data( [
+            'linux/integration-test.py',
+            'linux/output_checker.py',
+          ],
+          install_dir: get_option('prefix') / get_option('libexecdir') / 'upower'
+        )
+        install_subdir('linux/tests/',
+          install_dir: get_option('prefix') / get_option('libexecdir') / 'upower'
+        )
+
+        configure_file(
+          input: 'upower-integration.test.in',
+          output: 'upower-integration.test',
+          install_dir: get_option('datadir') / 'installed-tests' / 'upower',
+          configuration: cdata
+        )
+    endif
 endif
-- 
2.47.1

